Cabal/Distribution/Simple/Utils.hs

```diff
@@ -437,9 +439,41 @@ wrapText = unlines
 -- | Wraps text unless the @+nowrap@ verbosity flag is active
 wrapTextVerbosity :: Verbosity -> String -> String
 wrapTextVerbosity verb
-  | isVerboseNoWrap verb = unlines . lines -- makes sure there's a trailing LF
+  | isVerboseNoWrap verb = unlines . lines
   | otherwise            = wrapText
 
+-- | Convenience helper combining 'formatLogMsg' with 'hFlush'
+hPutLogMsg :: Handle -> Verbosity -> String -> NoCallStackIO ()
+hPutLogMsg handle verbosity msg = do
+    hPutStr handle =<< formatLogMsg verbosity msg
+    hFlush handle
+
+-- | Convenience helper combining 'formatMsgNoWrap' with 'hFlush'
+hPutLogMsgNoWrap :: Handle -> Verbosity -> String -> NoCallStackIO ()
+hPutLogMsgNoWrap handle verbosity msg = do
+    hPutStr handle =<< formatMsgNoWrap verbosity msg
+    hFlush handle
+
+-- | Prepend timestamp and/or wrap log message depending on
+-- 'Verbosity' settings.
+formatLogMsg :: Verbosity -> String -> NoCallStackIO String
+formatLogMsg verbosity = formatMsgNoWrap verbosity . wrapTextVerbosity verbosity
+
+-- | Prepends timestamp when the @+timestamp@ verbosity flag is active
```

Comment:
This is a bad comment, it says what the function does, as opposed to *when it should be called.*